gateway/mw_oas_validate_request.go

```diff
@@ -61,6 +63,8 @@ func (k *ValidateRequest) ProcessRequest(w http.ResponseWriter, r *http.Request,
 		errResponseCode = validateRequest.ErrorResponseCode
 	}
 
+	openapi3.SchemaErrorDetailsDisabled = validateRequest.SchemaErrorDetailsDisabled
```

Comment:
this is a global variable in `openapi3` isn't there a chance that this could affect other apis as well?

Reply:
As it is set in every api call before validation, each api will set its value to that and it won’t affect other APIs, right?

Regarding API creation, this is request validation. I couldn’t make sure if the object validation used the same flag for that. Let’s check.

Reply:
> As it is set in every api call before validation, each api will set its value to that and it won’t affect other APIs, right?
`openapi3.SchemaErrorDetailsDisabled` is not reset back to previous value. even if we do, concurrent requests are not protected right?

Reply:
Yes, there may be concurrency issue here. I didn't want to add a mutex directly. The library also doesn't provide it to us. It seems we should a mutex right? Or any other solution suggestion?

Reply:
Another option is to make this a gateway level config, thereby the behaviour stays consistent across all APIs + control APIs